src/main/java/mops/module/services/AntragService.java

```diff
@@ -140,4 +142,54 @@ public Modul approveModulCreationAntrag(Antrag antrag) {
     public Antrag getAntragById(Long id) {
         return antragRepository.findById(id).orElse(null);
     }
+
+    /**
+     * Gibt alle Versionen für das Modul mit der angebenen id als Liste aus.
+     * @param id Die id eines in der Datenbank existierenden (d.h. genehmigten) Moduls.
+     * @return Eine Liste aller Versionen des Moduls, wobei der erste Listeneintrag das Modul
+     *         im initialen Erstellungsantrag ist und der letzte Listeneintrag das Modul im
+     *         aktuellen Status ist. Da der Sichtbarkeitsstatus nicht über Anträge funktioniert,
+     *         ist die Sichtbarkeit aller Versionen immer null.
+     */
+    public LinkedList<Modul> getAllVersionsOfModulOldestFirst(Long id) {
+        List<Antrag> relevantApprovedAntraege = getAllApprovedAntraegeForModulOldestFirst(id);
+        LinkedList<Modul> modulVersions = new LinkedList<>();
+        for (int i = 1; i < relevantApprovedAntraege.size(); i++) {
+            //initiale Version (Creation-Antrag)
+            Modul versionI = JsonService.jsonObjectToModul(
+                    relevantApprovedAntraege.get(0).getJsonModulAenderung());
+            //für Schleifendurchgang i jeweils alle Modification-Antraege bis zum i-ten anwenden
+            for (int j = 0; j < i; j++) {
+                ModulService.applyAntragOnModul(versionI, relevantApprovedAntraege.get(j));
+            }
+            modulVersions.add(versionI);
+        }
+        //aktuelle Version
+        Modul currentModul = modulSnapshotRepository.findById(id).orElse(null);
```

Comment:
Steht das aktuelle Modul dann nicht doppelt in der Liste?

Reply:
Nein, da innere Schleife immer eins niedriger als äußere zählt.